File: UnrealEngine/Engine/Plugins/Experimental/Mutable/Source/MutableRuntime/Private/MuR/Memory.cpp
// Copyright Epic Games, Inc. All Rights Reserved.

#include "MuR/BlockCompression/Miro/Miro.h"
#include "MuR/MutableMemory.h"

namespace mu
{
	static int s_initialized = 0;
	static int s_finalized = 0;

	void Initialize()
	{
		if (!s_initialized)
		{
			s_initialized = 1;
			s_finalized = 0;

			miro::initialize();
		}
	}

	void Finalize()
	{
		if (s_initialized && !s_finalized)
		{
			miro::finalize();

			s_finalized = 1;
			s_initialized = 0;
		}
	}

}

File: UnrealEngine/Engine/Source/Programs/Shared/EpicGames.Tracing/UnrealInsights/Events/Memory.cs
// Copyright Epic Games, Inc. All Rights Reserved.

using System;
using System.Collections.Generic;
using System.IO;

namespace EpicGames.Tracing.UnrealInsights.Events
{
	public class MemoryMemoryScopeEvent : ITraceEvent
	{
		public static readonly EventType EventType = new EventType(0, "Memory", "MemoryScope", EventType.FlagNone,
			new List<EventTypeField>() { new EventTypeField(0, 4, EventTypeField.TypeInt32, "Id") });

		public ushort Size => throw new NotImplementedException();
		public EventType Type => EventType;
		
		public void Serialize(ushort Uid, BinaryWriter Writer)
		{
			throw new NotImplementedException();
		}
	}
}